tough/src/lib.rs

```diff
@@ -266,28 +271,174 @@ impl<'a, T: Transport> Repository<'a, T> {
+    fn snapshot_filename(&self) -> String {
+        if self.root.signed.consistent_snapshot {
+            format!("{}.snapshot.json", self.snapshot.signed.version)
+        } else {
+            "snapshot.json".to_owned()
+        }
+    }
+
+    /// Prepends the version number to the targets.json filename if using consistent snapshot mode.
+    fn targets_filename(&self) -> String {
+        if self.root.signed.consistent_snapshot {
+            format!("{}.targets.json", self.targets.signed.version)
+        } else {
+            "targets.json".to_owned()
+        }
+    }
+
+    /// Prepends the version number to the timestamp.json filename if using consistent snapshot mode.
+    fn timestamp_filename(&self) -> String {
+        if self.root.signed.consistent_snapshot {
+            format!("{}.timestamp.json", self.snapshot.signed.version)
+        } else {
+            "timestamp.json".to_owned()
+        }
+    }
+
+    /// Prepends the target digest to the name if using consistent snapshots. Returns both the
+    /// digest and the filename.
+    fn target_digest_and_filename(&self, target: &Target, name: &str) -> (Vec<u8>, String) {
+        let sha256 = &target.hashes.sha256.clone().into_vec();
+        if self.consistent_snapshot {
+            (sha256.clone(), format!("{}.{}", hex::encode(sha256), name))
+        } else {
+            (sha256.clone(), name.to_owned())
+        }
+    }
 
-            Some(fetch_sha256(
-                self.transport,
-                self.targets_base_url.join(&file).context(error::JoinUrl {
-                    path: file,
+    /// Fetches the signed target using `Transport`. Aborts with error if the fetched target is
+    /// larger than its signed size.
+    fn fetch_target(&self, target: &Target, digest: &[u8], filename: &str) -> Result<impl Read> {
+        fetch_sha256(
+            self.transport,
+            self.targets_base_url
+                .join(&filename)
+                .context(error::JoinUrl {
+                    path: filename,
                     url: self.targets_base_url.to_owned(),
                 })?,
-                target.length,
-                "targets.json",
-                sha256,
-            )?)
-        } else {
-            None
-        })
+            target.length,
+            "targets.json",
+            digest,
+        )
+    }
+
+    /// Copies a file using `Transport` to `outdir`.
+    fn cache_file_from_transport<P: AsRef<Path>>(&self, filename: &str, outdir: P) -> Result<()> {
+        let mut read = fetch_max_size(
+            self.transport,
+            self.metadata_base_url
+                .join(filename)
+                .context(error::JoinUrl {
+                    path: filename,
+                    url: self.metadata_base_url.to_owned(),
+                })?,
+            self.limits.max_root_size, // TODO(webern) - this is not correct
```

Comment:
I discovered this bug after opening the PR. I will fix this. It should be max root size for root.json files and signed size (i think) for other metadata file. So probably max_size needs to be added to the function signature.